src/Microsoft.AspNetCore.Mvc.Formatters.Xml/ProblemDetailsWrapper.cs

```diff
@@ -0,0 +1,161 @@
+        }
+
+        internal static void ReadProperty(XmlReader reader, ProblemDetails problemDetails, string key)
+        {
+            var value = reader.ReadInnerXml();
+
+            if (key == nameof(problemDetails.Detail))
+            {
+                problemDetails.Detail = value;
+            }
+            else if (key == nameof(problemDetails.Instance))
+            {
+                problemDetails.Instance = value;
+            }
+            else if (key == nameof(problemDetails.Status))
+            {
+                problemDetails.Status = string.IsNullOrEmpty(value) ?
+                    (int?)null :
+                    int.Parse(value, CultureInfo.InvariantCulture);
+            }
+            else if (key == nameof(problemDetails.Title))
+            {
+                problemDetails.Title = value;
+            }
+            else if (key == nameof(problemDetails.Type))
+            {
+                problemDetails.Type = value;
+            }
+            else
+            {
+                problemDetails.ExtensionMembers.Add(key, value);
+            }
+        }
+
+        /// <inheritdoc />
+        public void WriteXml(XmlWriter writer) => WriteProblemDetails(writer, ProblemDetails);
+        
+        internal static void WriteProblemDetails(XmlWriter writer, ProblemDetails problemDetails)
+        {
+            if (!string.IsNullOrEmpty(problemDetails.Detail))
+            {
+                writer.WriteElementString(
+                    XmlConvert.EncodeLocalName(nameof(problemDetails.Detail)),
+                    problemDetails.Detail);
+            }
+
+            if (!string.IsNullOrEmpty(problemDetails.Instance))
+            {
+                writer.WriteElementString(
+                    XmlConvert.EncodeLocalName(nameof(problemDetails.Instance)),
+                    problemDetails.Instance);
+            }
+
+            if (problemDetails.Status.HasValue)
+            {
+                writer.WriteStartElement(XmlConvert.EncodeLocalName(nameof(problemDetails.Status)));
+                writer.WriteValue(problemDetails.Status.Value);
+                writer.WriteEndElement();
+            }
+
+            if (!string.IsNullOrEmpty(problemDetails.Title))
+            {
+                writer.WriteElementString(
+                    XmlConvert.EncodeLocalName(nameof(problemDetails.Title)),
+                    problemDetails.Title);
+            }
+
+            if (!string.IsNullOrEmpty(problemDetails.Type))
+            {
+                writer.WriteElementString(
+                    XmlConvert.EncodeLocalName(nameof(problemDetails.Type)),
+                    problemDetails.Type);
+            }
+
+            foreach (var keyValuePair in problemDetails.ExtensionMembers)
+            {
+                var key = keyValuePair.Key;
+                var value = keyValuePair.Value;
+
+                writer.WriteStartElement(XmlConvert.EncodeLocalName(key));
```

Comment:
Don't see anything preventing an empty key in `ExtensionMembers` and suggest adding `EmptyKey` handling in this class.